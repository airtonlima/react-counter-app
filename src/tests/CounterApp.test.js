import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import CounterApp from '../CounterApp';

describe('Pruebas en <CounterApp />', () => {

    let wrapper = shallow( <CounterApp /> );

    beforeEach(() => {

        wrapper = shallow( <CounterApp /> );
    });
    test('#01 - Debe mostrar <CounterApp /> correctamente', () => {

        expect(wrapper).toMatchSnapshot();
    });

    test('#02 - Debe mostrar el valor por defecto de 100', () => {

        const wrapper = shallow(<CounterApp value={ 100 } />);

        const valueParrafo = wrapper.find('h2').text().trim();

        expect(valueParrafo).toBe('100');
    });

    test('#03 - Debe incrementar con el botón +1', () => {
        
        wrapper.find('button').at(0).simulate('click');

        const counterText = wrapper.find('h2').text().trim();

        expect(counterText).toBe('11');
    });

    test('#04 - Debe incrementar con el botón -1', () => {
        
        wrapper.find('button').at(2).simulate('click');

        const counterText = wrapper.find('h2').text().trim();

        expect(counterText).toBe('9');
    });

    test('#05 - Debe de colocar el valor por defecto con el botón Reset', () => {
        
        const wrapper = shallow( <CounterApp value={ 105 } />);
        
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(1).simulate('click');

        const counterText = wrapper.find('h2').text().trim();

        expect( counterText ).toBe('105');
    });

});
