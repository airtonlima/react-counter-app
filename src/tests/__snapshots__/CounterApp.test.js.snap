// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> #01 - Debe mostrar <CounterApp /> correctamente 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
     Reset
  </button>
  <button
    onClick={[Function]}
  >
     -1
  </button>
</Fragment>
`;

exports[`Pruebas en <CounterApp /> #03 - Debe incrementar con el bot√≥n +1 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
     Reset
  </button>
  <button
    onClick={[Function]}
  >
     -1
  </button>
</Fragment>
`;

exports[`Pruebas en <CounterApp /> #03 - Debe mostrar <CounterApp /> correctamente 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
     Reset
  </button>
  <button
    onClick={[Function]}
  >
     -1
  </button>
</Fragment>
`;
